mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs

```diff
@@ -916,10 +917,95 @@ class PacketReader {
 					return new ValueImpl () { Type = etype, Id = ReadId () };
 				case (ElementType)ValueTypeId.VALUE_TYPE_ID_PARENT_VTYPE:
 					return new ValueImpl () { Type = etype, Index = ReadInt () };
+				case (ElementType)ValueTypeId.VALUE_TYPE_ID_FIXED_ARRAY:
+					return ReadValueFixedSize ();
 				default:
 					throw new NotImplementedException ("Unable to handle type " + etype);
 				}
 			}
+			
+			public ValueImpl ReadValueFixedSize () {
+				var lenFixedSize = 1;
+				ElementType etype = (ElementType)ReadByte ();
+				if (connection.Version.AtLeast (2, 53)) {
+					lenFixedSize = ReadInt ();
```

Comment:
Older protocol versions can't send the type code which would cause this method to be reached.